src/main/java/com/google/planningworks/cdap/plugins/sink/AnaplanSink.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.planningworks.cdap.plugins.sink;
+
+import com.anaplan.client.AnaplanService;
+import com.anaplan.client.AnaplanService.AnaplanConfig;
+import com.anaplan.client.ex.ServerFilesNotFoundException;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema.Field;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.batch.SparkExecutionPluginContext;
+import io.cdap.cdap.etl.api.batch.SparkPluginContext;
+import io.cdap.cdap.etl.api.batch.SparkSink;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.stream.Collectors;
+import org.apache.spark.api.java.JavaRDD;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * CDF Sink plugin for importing data to Anaplan server via the wrapper {@link AnaplanService} of
+ * Anaplan API.
+ */
+@Plugin(type = SparkSink.PLUGIN_TYPE)
+@Name(AnaplanSink.PLUGIN_NAME)
+@Description("Import data into Anaplan model.")
+public class AnaplanSink extends SparkSink<StructuredRecord> {
+
+  public static final String PLUGIN_NAME = "AnaplanSink";
+  private static final int THREAD_NUM = 1; // Hardcoded to 1 for working with Anaplan REST API 2.0
+
+  private static final Logger LOG = LoggerFactory.getLogger(AnaplanSink.class);
+
+  private final AnaplanSinkConfig config;
+
+  public AnaplanSink(AnaplanSinkConfig config) {
+    this.config = config;
+  }
+
+  /**
+   * Configures the pipeline and validates the configurations when the pipeline is deployed. If this
+   * throws an exception, the pipeline will not be deployed and the user will be shown the error
+   * message.
+   *
+   * @param pipelineConfigurer the collection of the pipeline configuration and context
+   */
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    super.configurePipeline(pipelineConfigurer);
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    config.validate(collector);
```

Comment:
it would also be good to validate that all the fields in the input schema are supported types (assuming schema is not null). The input schema can be fetched from the StageConfigurer.

Does anaplan have any restrictions on column names? If so, would be good to validate the field names too.

Reply:
Yeah, added. Thank you very much for the suggestion.

Checked with Anaplan team, no restriction on the column name as long as not character out of the ASCII range. Quite relaxing.